controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java

```diff
@@ -132,54 +131,9 @@ public void createScope(final CreateScopeRequest createScopeRequest, final Secur
         .thenAccept(x -> LoggerHelpers.traceLeave(log, "createScope", traceId));
     }
 
-    private void authenticateAuthorize(String resourceName, AuthHandler.Permissions level) throws AuthException {
-        if (pravegaAuthManager != null ) {
-            List<String> authParams = headers.getRequestHeader(HttpHeaders.AUTHORIZATION);
 
-            if (authParams == null || authParams.isEmpty()) {
-                throw new AuthenticationException("Auth failed for " + resourceName);
-            }
-
-            String credentials = authParams.get(0);
-            Preconditions.checkNotNull(credentials, "Credentials not specified in the parameters.");
-
-            if (!pravegaAuthManager.authenticateAndAuthorize(resourceName, credentials, level)) {
-                throw new AuthorizationException("Auth failed for " + resourceName, Status.FORBIDDEN.getStatusCode());
-            }
-        }
-    }
-
-    private Principal authenticate() throws AuthException {
-        if (pravegaAuthManager != null ) {
-            List<String> authParams = headers.getRequestHeader(HttpHeaders.AUTHORIZATION);
-
-            if (authParams == null || authParams.isEmpty()) {
-                throw new AuthenticationException("Auth failed.");
-            }
-
-            String credentials = authParams.get(0);
-            Preconditions.checkNotNull(credentials, "Credentials not specified in the parameters.");
-
-            return pravegaAuthManager.authenticate(credentials);
-        }
-        return null;
-    }
-
-    private void authorize(String resourceName, Principal principal, AuthHandler.Permissions level) throws AuthException {
-        if (pravegaAuthManager != null ) {
-            List<String> authParams = headers.getRequestHeader(HttpHeaders.AUTHORIZATION);
-
-            if (authParams == null || authParams.isEmpty()) {
-                throw new AuthenticationException("Auth failed for " + resourceName);
-            }
-
-            String credentials = authParams.get(0);
-            Preconditions.checkNotNull(credentials, "Credentials not specified in the parameters.");
-
-            if (!pravegaAuthManager.authorize(resourceName, principal, credentials, level)) {
-                throw new AuthorizationException("Auth failed for " + resourceName, Status.FORBIDDEN.getStatusCode());
-            }
-        }
+    private List<String> fetchAuthorizationHeader() {
```

Comment:
`getAuthorizationHeader` sounds more appropriate

Reply:
Yeah, I initially thought the logic it contains is expensive. I avoided calling it ``get..()`` as one usually expects that it is cheap to call get...() as the return value is say cached/precomputed, and I thought that is not the case here. On a deeper look, it seems like the method really meets the criteria. Have renamed it to get..().